packages/rehype-mathjax/chtml.d.ts

```diff
@@ -16,6 +16,28 @@ interface MathJaxCHtmlOptions {
   adaptiveCSS?: boolean
 }
 
-declare const renderCHtml: Plugin<[MathJaxCHtmlOptions]>
+// http://docs.mathjax.org/en/latest/options/input/tex.html#the-configuration-block
+interface MathJaxInputTexOptions {
+  packages: string[]
+  inlineMath: [[string, string]]
+  displayMath: [[string, string]]
+  processEscapes: boolean
+  processEnvironments: boolean
+  processRefs: boolean
+  digits: RegExp
+  tags: 'none' | 'ams' | 'all'
+  tagSide: 'left' | 'right'
+  tagIndent: string
+  useLabelIds: boolean
+  multlineWidth: string
+  maxMacros: number
+  maxBuffer: number
+  baseURL: string
+  formatError: (jax: any, err: any) => string
+}
```

Comment:
do the Math Jax typings have an interface for this that could be reused?
Could these easily be upstreamed to MathJax?

Reply:
There isn't a typing that could be reused. I doubt that this could be upstreamed to MathJax because of its component architecture. Internally, MathJax uses type `OptionList = {[name: string]: any}` (from [Options.ts](https://github.com/mathjax/MathJax-src/blob/6ba6cca458080ec2a73c60f8b19db6491985f92b/ts/util/Options.ts)) pretty much everywhere for configuration, probably because the actual type for the options is determined dynamically.

There is validation in MathJax that throws for unhandled keys in the options (for instance, it throws if `tex` is a key for the CHTML output processor). I don't know specifically where that validation takes place, but it seems to me like a proper type for the options cannot be inferred statically. Inferring the type would require merging the various option types read from the list of packages configured in the input processors. That could be done if there were some schema object for the options of each component, in a similar fashion as how type inferrence works with [Yup](https://github.com/jquense/yup/blob/master/docs/typescript.md).